### ISSUE TITLE
[Feature]: Labelled entities in Genesis

### ISSUE THREAD

**Issue by MRiabov** [Genesis-Embodied-AI/Genesis#1410]
### What feature or enhancement are you proposing?

Hello,
As somebody who has to manage many Genesis entities and store them for longer use (think having multiple abstraction layers), I currently have to manage them through a separate `gs_entities` dict, which has to appear in 108 places in my codebase.
This isn't per se a problem, however it would be more convenient if such entities could be stored in `genesis.Scene` under labels, and `Scene.entities` would return a dict instead of a list.

This would simplify:
1. Getting link ids for constraints
2. Getting positions of bodies
3. Getting AABBs of bodies,
4. Etc. that uses `entity`.

### Motivation

As a RL researcher developing on top of Genesis and having scenes with 5+ bodies,
I want to make managing bodies simpler and take less space in my code,
Because this leads to better code and faster development cycles.

I would also add that I use `label` to denote types of bodies I'm managing. E.g. this loop:
```
for full_name, entity in gs_entities.items():
    part_name, part_type = full_name.split("@")
        if part_type in ("solid", "fixed_solid", "connector"):
             ...
       elif part_type == "control":
            ...
        elif part_type == "fastener":
            fastener_pos = torch.tensor(entity.get_pos(env_idx), device=device)
             ...
```
Maybe this means that adding some form of groups/auxillary data to entities would be desirable (however this can be easily managed by the user). Nevertheless, entity naming is already a great step to it.

### Potential Benefit

Faster development cycles.

### What is the expected outcome of the implementation work?

- `gs.add_entity()` would now have an optional `label` attribute.
- If label is added, it is possible to get it from `entities`, or `scene.get_entity()`<- `get_entity` would minimize changes necessary in codebase. 
- `label` can be queried the entity.

### Additional information

`gs_entities` appears in 108 places in my code.
<img width="345" height="547" alt="Image" src="https://github.com/user-attachments/assets/c023939b-aef5-482a-a5d2-397398136a40" />

I don't expect that this change should impact anything but providing an API for getting bodies and their labels.

**Comment by duburcqa** [comment]
Fixed by https://github.com/Genesis-Embodied-AI/Genesis/pull/2303
